ReactiveCocoaTests/Swift/SignalSpec.swift

```diff
@@ -453,6 +473,33 @@ class SignalSpec: QuickSpec {
 				observer.sendNext(1)
 				expect(lastValue).to(equal("2"))
 			}
+
+			it("should not keep resulting signal alive indefinitely") {
+				weak var signal: Signal<AnyObject, NoError>? = Signal.never.map { $0 }
+
+				expect(signal).to(beNil())
+			}
+
+			it("should not keep resulting signal alive indefinitely even if it has an observer") {
+				weak var signal: Signal<AnyObject, NoError>? = {
+					let signal: Signal<AnyObject, NoError> = Signal.never.map { $0 }
+					signal.observe(Observer())
+					return signal
+				}()
+				expect(signal).to(beNil())
+			}
+
+			it("should not keep resulting signal alive indefinitely even if it has an observer with retained disposable") {
```

Comment:
This test is different because unlike the previous one it has a disposable from `observe` retained.
Unlike such tests without `map` they both failed before the fix though, because `map` already retains the disposable.
This is how tests were prior to the fix:
init:
✅should not keep signal alive indefinitely
✅should not keep signal alive indefinitely even if it has an observer
❌should not keep signal alive indefinitely even if it has an observer with retained disposable
map:
❌should not keep resulting signal alive indefinitely
❌should not keep resulting signal alive indefinitely even if it has an observer
❌should not keep resulting signal alive indefinitely even if it has an observer with retained disposable
I've added all variations for `map` for completeness, but if you think it's excessive, I can leave only one.